pkg/operator/controllers/monitoring/monitoring_controller_test.go

```diff
@@ -111,11 +100,73 @@ alertmanagerMain:
   extraField: yeet
 prometheusK8s:
   extraField: prometheus
+`,
+		},
+		{
+			name: "empty volumeClaimTemplate struct is cleared out",
+			reconciler: func() *Reconciler {
+				return &Reconciler{
+					kubernetescli: fake.NewSimpleClientset(&corev1.ConfigMap{
+						ObjectMeta: cmMetadata,
+						Data: map[string]string{
+							"config.yaml": `
+alertmanagerMain:
+  volumeClaimTemplate: {}
+  extraField: alertmanager
+prometheusK8s:
+  volumeClaimTemplate: {}
+  bugs: not-here
+`,
+						},
+					}),
+					log:        log,
+					jsonHandle: new(codec.JsonHandle),
+				}
+			},
+			wantConfig: `
+alertmanagerMain:
+  extraField: alertmanager
+prometheusK8s:
+  bugs: not-here
+`,
+		},
+		{
+			name: "empty volumeClaimTemplate cleared out part 2",
```

Comment:
Unless I miss something, `"empty volumeClaimTemplate cleared out part 2"` case covers everything we need. We can probably drop `"empty volumeClaimTemplate struct is cleared out"` case. What do you think?

```suggestion
			name: "empty volumeClaimTemplate cleared out",
```

Reply:
forgot to remove that one - thought I hit an edge case so I threw in another additional test.  
Good catch will update.